### ISSUE TITLE
Raise OpenEXR/Imath minimum to 3.1 for OIIO 3.0

### ISSUE THREAD

**Issue by lgritz** [AcademySoftwareFoundation/OpenImageIO#4156]
A non-comprehensive list of sub-tasks, not necessarily in order:

- [ ] INSTALL.md -- document the new minimums
- [ ] externalpackages.cmake -- enforce the new minimums
- [ ] FindOpenEXR.cmake -- is this no longer needed because we can rely on OpenEXR's and Imath's exported configs?
- [ ] Search for OPENEXR_VERSION through the whole codebase, look for places where we are conditionally selecting code.
- [ ] ci.yml -- remove testing cases of the versions, and/or change those test cases to newer versions.
- [ ] `openexr.imageio/*.{cpp,h}` -- clean up everything where we special-case pre-3.1

We should not merge these fixes into the main branch until after April 1.
